x/upgrade/README.md

```diff
@@ -2,7 +2,17 @@
 
 ## Abstract
 
-The upgrade module removes the entrypoints to the standard upgrade module by not
-registering a message server. It registers the standard upgrade module types to
-preserve the ability to marshal them. Note that the keeper of the standard
-upgrade module is still added to the application.
+This upgrade module is a fork of the cosmos-sdk's
+[x/upgrade](https://github.com/cosmos/cosmos-sdk/tree/main/x/upgrade) module
+that removes the entrypoints to the standard upgrade module by not registering a
+message server. It registers the standard upgrade module types to preserve the
+ability to marshal them. Note that the keeper of the standard upgrade module is
+still added to the application.
```

Comment:
It might be nice to add the motivation around this. 

I'm also a little unclear why we need to remove the message server. Only `x/gov` can invoke the methods right? Which we've removed the handler for

Reply:
this is a good point, we did remove the handler's so it should be sufficient to simply remove those handlers. I think the thinking was something along the lines of "why keep code that we can't access", and it was within reason to remove it entirely.

Reply:
+1 I think motivating context is in https://github.com/celestiaorg/celestia-app/pull/1491/files#diff-2441f8dbdfce977902fdfe6f74a28966c7b4202c387ee5d0999e394216dc013cR56-R59 so will try to copy some of that content in.

I kinda wish we merged ADR-016 considered we implemented option 2.

Reply:
https://github.com/celestiaorg/celestia-app/pull/2072/commits/2d0503e33b9950937ddfe67f4f1b30f49faf3145